src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerOptions.cs

```diff
@@ -63,5 +63,42 @@ public ConsoleLoggerFormat Format
         /// </summary>
         [System.ObsoleteAttribute("ConsoleLoggerOptions.UseUtcTimestamp has been deprecated. Use ConsoleFormatterOptions.UseUtcTimestamp instead.")]
         public bool UseUtcTimestamp { get; set; }
+
+        private ConsoleLoggerBufferFullMode _bufferFullMode = ConsoleLoggerBufferFullMode.Wait;
+        /// <summary>
+        /// Gets or sets the desired console logger behavior when buffer becomes full. Defaults to <c>Wait</c>.
+        /// </summary>
+        public ConsoleLoggerBufferFullMode BufferFullMode
+        {
+            get => _bufferFullMode;
+            set
+            {
+                if (value != ConsoleLoggerBufferFullMode.Wait && value != ConsoleLoggerBufferFullMode.DropWrite)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} is not a supported buffer mode value.");
+                }
+                _bufferFullMode = value;
+            }
+        }
+
+        internal const int DefaultMaxQueueLengthValue = 1024 * 1024;
```

Comment:
2500 is not enough for a default value. 

e.g. with

```c#
for (int i = 0; i < 10000; i++)
{
    logger.LogInformation(_1000Chars);
}
```

When testing to set the max size for the above `for` loop, the max size increase to `2500` won't be enough but for the same use case setting to 10,000 arbitrarily, makes sure we don't hit the lock contention and the speed jump that we want happens too.

Reply:
I have no strong opinion of value default. Just sharing my observation here in case it helps decide on the best number to choose. Changed default from 1024*1024 to 2500 in latest commit.

cc @davidfowl